src/ipc/capnp/common-types.h

```diff
@@ -6,16 +6,63 @@
 #define BITCOIN_IPC_CAPNP_COMMON_TYPES_H
 
 #include <clientversion.h>
+#include <protocol.h>
 #include <streams.h>
 #include <univalue.h>
+#include <util/result.h>
+#include <util/translation.h>
 
 #include <cstddef>
 #include <mp/proxy-types.h>
 #include <type_traits>
+#include <unordered_set>
 #include <utility>
 
 namespace ipc {
 namespace capnp {
+//! Convert kj::ArrayPtr to base_blob.
+template <typename T>
+inline T ToBlob(const kj::ArrayPtr<const kj::byte>& array)
+{
+    return T({array.begin(), array.begin() + array.size()});
+}
+
+//! Convert base_blob to kj::ArrayPtr.
+template <typename T>
+inline kj::ArrayPtr<const kj::byte> ToArray(const T& blob)
+{
+    return {reinterpret_cast<const kj::byte*>(blob.data()), blob.size()};
+}
+
+//! Construct a ParamStream wrapping data stream with serialization parameters
+//! needed to pass transaction and address objects between bitcoin processes.
+template<typename S>
+auto Wrap(S& s)
+{
+    return ParamsStream{s, TX_WITH_WITNESS, CAddress::V2_NETWORK};
```

Comment:
i think `Wrap` could be parameterized with any `CAddress` version.
e.g useful if someone wanna test the native crypto noise framework between bitcoin processes.
or in the future experiment with better crypto-primitives for bip324.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29409#discussion_r1515411394

> i think `Wrap` could be parameterized with any `CAddress` version. e.g useful if someone wanna test the native crypto noise framework between bitcoin processes. or in the future experiment with better crypto-primitives for bip324.

For now the Wrap function is basically a workaround for not having a serialization option that says "serialize all the data in this object to a stream, I don't care about the specific serialization format." So the formats here just chosen to preserve all data, and don't really matter otherwise.

But I could imagine use-cases like you are suggesting where we may want to customize which serialization format is used depending on the IPC connection. I think it would not be that hard to do by adding parameters to Wrap, which is not called many places, or by writing different CustomReadField/BuildField overloads which do not call Wrap. For now though the easiest thing is for the wrapped stream to just hardcode parameters that serialize all the data.